sys/include/net/nanocoap.h

```diff
@@ -714,6 +714,55 @@ static inline void coap_hdr_set_type(coap_hdr_t *hdr, unsigned type)
     hdr->ver_t_tkl &= ~0x30;
     hdr->ver_t_tkl |= type << 4;
 }
+
+/**
+ * @brief       Get the token length of a CoAP over UDP (DTLS) packet
+ * @param[in]   hdr     CoAP over UDP header
+ * return       The size of the token in bytes
+ *
+ * @warning     This API is super goofy. It assumes that the packet is valid
+ *              and will read more than `sizeof(*hdr)` into the data `hdr`
+ *              points to while crossing fingers hard.
+ *
+ * @deprecated  This function was introduced to keep legacy code alive.
+ *              Introducing new callers should be avoided. In the RX path an
+ *              @ref coap_pkt_t will be available, so that you can call
+ *              @ref coap_get_token instead. In the TX path the token was
+ *              added by us, so we really should know.
+ */
+static inline size_t coap_hdr_get_token_len(const coap_hdr_t *hdr)
+{
+    const uint8_t *buf = (const void *)hdr;
+    switch (coap_hdr_tkl_ext_len(hdr)) {
+    case 0:
+        return hdr->ver_t_tkl & 0xf;
+    case 1:
+        return buf[sizeof(coap_hdr_t)] + 13;
+    case 2:
+        return byteorder_bebuftohs(buf + sizeof(coap_hdr_t)) + 269;
```

Comment:
why not `13*13+100`?

Reply:
or `13*13+13*7+9`?

Reply:
That constant has no name in the RFC as well:

> 13: An 8-bit unsigned integer directly precedes the "Token" field and indicates the length of the "Token" field minus 13.
> 14: A 16-bit unsigned integer in network byte order directly precedes the "Token" field and indicates the length of the "Token" field minus 269.

I think it is actually easier to not introduce a RIOT specific name to hide the numeric value when the RFC has no name for it, as you would grep for `269` when looking a the RFC.

Reply:
I added a comment

Reply:
13 + UINT8_MAX = 268 < 269. It reminds me of  extended options. It does not have  a name because that's how the encoding works mathematically.

Reply:
It's 14 (value of `tkl` field) + 255 (max value we could represent with one extended token length byte, but since we use two bytes we already exceed that, so make that implicit)